packages/eslint-plugin-office-addins/src/rules/test-for-null-using-isNullObject.ts

```diff
@@ -129,7 +129,7 @@ export = {
               fix: function (fixer: RuleFixer) {
                 var ruleFix: RuleFix;
                 if (
-                  isInBinaryNullTest(reference.identifier) &&
+                  isInBinaryNullTest(reference.identifier as TSESTree.Identifier) &&
```

Comment:
Here you could test like:
```
if (reference.identifier.type === TSESTree.Identifier
&& isInBinaryNullTest(reference.identifier)
```

Reply:
This is a technicality with dependencies and the cast ensures the nest package is cast into the upper level package, though they are the same package.

Reply:
I know this code you made works on this case. But I think it might have some potential bugs like:
What if `reference.identifier` was not a `TSESTree.Identifier`? This explicit cast could make bad issues appear.

Reply:
Typescript types are what helps us make sure the types are correct.  The reason for the cast is because after adjusting the 
dependencies the compiler complains about:

eslint-plugin-office-addins/node_modules/@typescript-eslint/types/dist/ast-spec Identifier 
. . . not being the same as . . . 
eslint-plugin-office-addins/node_modules/@typescript-eslint/typescript-estree/node_modules/@typescript-eslint/types/dist/ast-spec

Note that they are the same, it just that one seems to be resolving to the package nested in typescript-estree and the other is not.